nemo_rl/algorithms/grpo.py

```
@@ -1690,16 +1706,29 @@ def async_grpo_train(
+                        assert full_metric_name.count(":") == 1, (
+                            "metric_name must contain exactly one colon"
+                        )
+                        parts = full_metric_name.split(":")
+                        train_or_val = "val" if "val" in parts[0] else "train"
+                        metric_name = parts[1]
+
+                        metrics_source = (
+                            metrics if train_or_val == "train" else val_metrics
+                        )
+                        if metric_name not in metrics_source:
                             warnings.warn(
```

Ruff (0.14.0)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)